build-clang-with-args.sh

```diff
@@ -50,8 +50,11 @@ cd "${build_top_dir}/build"
 
 
 llvm_dir="${build_top_dir}/build/llvm-project"
-git clone https://github.com/llvm/llvm-project.git "${llvm_dir}"
+if [ ! -d "${llvm_dir}" ]; then
+  git clone https://github.com/llvm/llvm-project.git "${llvm_dir}"
+fi
 cd "${llvm_dir}"
+git fetch origin
 git checkout --force "${LLVM_VERSION}"
```

Comment:
It can be a separate patch, but perhaps we could take the opportunity to do a shallow clone?

```
git init
git remote add origin https://github.com/llvm/llvm-project.git
git fetch origin <commit>
git reset --hard FETCH_HEAD
```

Reply:
Yeah, I thought about it and then went for the minimal solution. I'd prefer a short script/shell function to clone a repository in this way that we can then reuse in the multiple scripts where we need something similar.